src/interdiction/bilevel_adaptive_arc.jl

```diff
@@ -0,0 +1,134 @@
+  # first cut variables
+  @variable(second_level, δ[i = 1:n,j = 1:n] ≥ 0)
+  # second cut variables
+  @variable(second_level, σ[i = 1:n])
+  # linearization variables: v = δμ
+  @variable(second_level, ν[i = 1:n, j = 1:n])
+
+  # set objective
+  @objective(second_level, Min, sum{
+    x_value[i, j] * δ[i, j] - x_value[i, j] * ν[i, j], i = 1:n, j = 1:n;
+    capacity_matrix[i, j] > 0})
+
+  # constraints over the edges
+  for e in edges(flow_graph)
+    i, j = src(e), dst(e)
+    # first linearization constraint for v
+    @constraint(second_level, ν[i, j] ≤ μ[i, j])
+    # second linearization constraint for v
+    @constraint(second_level, ν[i, j] ≤ δ[i, j])
+    if i == source
+      # cut constraint for edge (i,j) when i is the source
+      @constraint(second_level, δ[i, j] + σ[j] ≥ 1 )
+    elseif j == target
+      # cut constraint for edge (i,j) when j is the destination
+      @constraint(second_level, δ[i, j] - σ[i] ≥ 0 )
+    else
+      # cut constraint for edge (i,j) in the remaining cases
+      @constraint(second_level, δ[i, j] + σ[j] - σ[i] ≥ 0)
+    end
+  end
+
+  # constraint on the upper bound k for the number of attacks
+  @constraint(second_level, sum{μ[i, j], i = 1:n, j = 1:n} ≤ attacks)
+
+  return second_level, δ, ν
+end
+
+function bilevel_adaptive_arc{T<:AbstractFloat}(
+  flow_graph::DiGraph,                          # the input graph
+  source::Int,                                  # the source vertex
+  target::Int,                                  # the target vertex
+  capacity_matrix::AbstractArray{T, 2},         # edge flow capacities
+  attacks::Int,                                 # argument for attacks
+  solver::AbstractMathProgSolver,               # keyword for solver
+  rtol::T,                                      # absolute tolerance
+  atol::T,                                      # relative tolerance
+  time_limit::Float64                           # time limit (seconds)
+  )
+	start_time = time()               # time stamp (seconds)
+  n = nv(flow_graph)                # size of the network
+  lower_bound = 0.
+  upper_bound = 0.
+
+  # Initialization : first level
+  first_level, x, z =
+    init_first_level_arc(flow_graph, source, target, capacity_matrix, solver)
+
+  # Loop over the first level while adding cuts from the second level
+  while solve(first_level) ≠ :Infeasible && time() - start_time < time_limit
+    # Store first level results
+    upper_bound, x_value = getobjectivevalue(first_level), getvalue(x)
+
+    # Initialization : second level
+    second_level, δ, ν =
+      init_second_level_arc(flow_graph, source, target, capacity_matrix,
+                                                     attacks, solver, x)
+
+    # Solve second level and update lower_bound
+		solve(second_level)
+		lower_bound = getobjectivevalue(second_level)
+
+    # If the bounds are tight, exit the loop
+    (isapprox(upper_bound, lower_bound, rtol = rtol, atol = atol)) && break
```

Comment:
OK, this indentation thing is really odd. It appears on GitHub but not on my editor. I will correct it tomorrow with using a basic text editor. Sorry for the delay.